File: 2232-adding-spaces-to-a-string/2232-adding-spaces-to-a-string.py
class Solution:
    def addSpaces(self, s: str, spaces: List[int]) -> str:
        ds=""
        prev=0
        spaces.append(len(s))
        for i in spaces:
            ds+=s[prev:i]+' '
            prev=i
        print(ds)
        return ds[0:-1]

File: 2232-adding-spaces-to-a-string/2232-adding-spaces-to-a-string.java
class Solution {
    public String addSpaces(String s, int[] spaces) {
        StringBuilder ds=new StringBuilder();
        int prev=0;
        for(int i:spaces){
            ds.append(s.substring(prev,i)).append(" ");
            prev=i;
        }
        ds.append(s.substring(prev));
        return ds.toString();
    }
}